### PR TITLE
Add ability to sync Plex Ratings to MediaMonkey

### LINKED ISSUE #7: Rating mismatches
Hi,

I thought I'd try this script out against a Test collection, just one album on the PLEX side and see what happened.

I started by:
Cleaning all the Ratings from the folder in PLEX
Cleaning all the Ratings on the folder in MM.
Setting one file to 5stars in MM
Syncing the tags between MM and the files.

Then ran the script.

It did match up the 5star test file I'd set, but it also set some other MP3s in PLEX to 2star and 3star, which aren't set in MM.

Whats the threshold it uses to commit to the update ?
When I look at the debug log for one of the wrong files:

	Line 851: [16:38:59] DEBUG: Found match with score 35.0 for Dido - No Angel - Hunter: Adam Carroll - Lookin' Out the Screen Door - Hunter's Song
	Line 5487: [16:41:50] DEBUG: Updating rating of track "Adam Carroll - Lookin' Out the Screen Door - Hunter's Song" to 3.0 stars

So I think its matching Dido there to Adam Carroll?

I'll look through the code next to see if I can spot a threshold value variable

### PR DESCRIPTION
**Pull request by smilerz**
-command line flag added to reverse direction of sync
-resolved this # TODO: use public accessor method, once PR-263 gets merged: https://github.com/pkkid/python-plexapi/pull/263
-fixes #7, bug causing false positives on rating mismatch

### CODE REVIEW

File: MediaPlayer.py [hunk 1]
```diff
@@ -152,12 +152,16 @@ def read_playlists(self):
 
 	def read_track_metadata(self, track):
 		tag = AudioTag(track.Artist.Name, track.Album.Name, track.Title)
-		tag.rating = self.get_normed_rating(track.Rating)
+		if track.Rating <0: 
+			tag.rating=0
+		else:
+			tag.rating = self.get_normed_rating(track.Rating)
```
Comment (patzm) [inline MediaPlayer.py]: sharing this logic with all consumers of `get_normed_rating` might make sense. Did you consider this? I.e. putting this condition inside the function implementation.

File: sync_pair.py [hunk 2]
```diff
@@ -104,54 +121,68 @@ def match(self, candidates=None, match_threshold=30):
...
+				self.source, self.rating_source, self.rating_destination)
+			)
 
 		return score
 
 	def resolve_conflict(self):
 		return NotImplemented
 
 	def similarity(self, candidate):
+        ########################i'm here
```
Comment (patzm) [inline sync_pair.py]: great that you are here :wink:

File: sync_ratings.py [hunk 3]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env python3
-
```
Comment (patzm) [inline sync_ratings.py]: why remove this? You can't execute it with `./sync_ratings.py` anymore then.

**Review by patzm — COMMENTED** [review]
@smilerz nice contribution! Thanks! I can't test this, because for me it was a one-way port. I have a few minor remarks. Ping me again, then we can merge :+1:

**Comment by patzm** [comment]
Oh, and I understand that this implements bi-directional sync. I suggest you edit the `Readme.md` as well (in https://github.com/patzm/plex-music-rating-sync#potential-next-features).
Also, should we upgrade the lower bound version for the plex API to reflect the merged PR?

**Comment by smilerz** [comment]
Couple notes:  
-Is it bi-directional?  Or one-directional with a choice on which direction to sync?
-I am currently using plexapi 4.2.0 how do you update the lower bound (README.md updated)?  is that in requirements.txt?
-Never done this before - do I just create a new pull request to get these changes over?

**Comment by patzm** [comment]
> Couple notes:
> -Is it bi-directional? Or one-directional with a choice on which direction to sync?

Oh true, it is one-directional, but you can now choose both directions. So maybe better not to update the readme. bi-directional involves many more features.

> -I am currently using plexapi 4.2.0 how do you update the lower bound (README.md updated)? is that in requirements.txt?

Yes, you update it here: https://github.com/patzm/plex-music-rating-sync/blob/8da87b82d9467b61aac30feb3403a34b9991c9dd/requirements.txt#L1

> -Never done this before - do I just create a new pull request to get these changes over?

Simply push the changes to the branch in your repo and this will appear here in the PR.

**Review by patzm — APPROVED** [review]

**Comment by patzm** [comment]
Can I merge? nice work again

**Comment by smilerz** [comment]
> Can I merge? nice work again

go for it - thanks!

**Pull request merged**